test/src/test/java/org/corfudb/integration/ServerRestartIT.java

```diff
@@ -462,6 +468,205 @@ public void sequencerTailsRecoveryTest() throws Exception {
         assertThat(shutdownCorfuServer(corfuServerProcess)).isTrue();
 
     }
+
+    /**
+     * If a tokenResponse is received in the previous epoch,
+     * a write should discard the tokenResponse and throw an exception.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void discardTokenReceivedInPreviousEpoch() throws Exception {
+        final int timeToWaitInSeconds = 3;
+
+        Process corfuServerProcess = runCorfuServer();
+
+        assertThat(shutdownCorfuServer(corfuServerProcess)).isTrue();
+
+        corfuServerProcess = runCorfuServer();
+
+        CorfuRuntime corfuRuntime = createDefaultRuntime();
+        TokenResponse tr = corfuRuntime.getSequencerView().nextToken(Collections.emptySet(), 1);
+
+        assertThat(tr.getEpoch() == 1);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Complete the assertion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2970)